ark/segmentation/marker_quantification.py

```diff
@@ -5,16 +5,155 @@
+
+    # get regionprop info needed for single compartment computations
+    props = regionprops(segmentation_labels)
+
+    # generate the required data for each cell
+    for prop in props:
+        for re in regionprops_single_comp:
+            cell_props_single_comp[re].append(REGIONPROPS_FUNCTION[re](prop, **kwargs))
+
+    # convert the dictionary to a DataFrame
+    cell_props_single_comp = pd.DataFrame.from_dict(cell_props_single_comp)
+
+    # append the single compartment derived props info to the cell_props DataFrame
+    cell_props = pd.concat([cell_props, cell_props_single_comp], axis=1)
+
+    return cell_props
+
+
+def assign_single_compartment_features(marker_counts, compartment, cell_props, cell_coords,
+                                       cell_id, label_id, input_images, regionprops_names,
+                                       extraction, **kwargs):
+    """Assign computed regionprops features and signal intensity to cell_id in marker_counts
+
+    Args:
+        marker_counts (xarray.DataArray):
+            xarray containing segmentaed data of cells x markers
+        compartment (str):
+            either 'whole_cell' or 'nuclear'
+        cell_props (pandas.DataFrame):
+            regionprops information for each cell
+        cell_coords (numpy.ndarray):
+            values representing pixels within one cell
+        cell_id (int):
+            id of the cell
+        label_id (int):
+            id used to index into cell_props
+        input_images (xarray.DataArray):
+            rows x columns x channels matrix of imaging data
+        regionprops_names (list):
+            all of the regionprops features (including derived, except nuclear-specific)
+        extraction (str):
+            the extraction method to use for signal intensity calculation
+        **kwargs:
+            arbitrary keyword arguments
+
+    Returns:
+        xarray.DataArray:
+            the updated marker_counts matrix with data for the specified cell_id and compartment
+    """
+
+    # get centroid corresponding to current cell
+    kwargs['centroid'] = np.array((
+        cell_props.loc[cell_props['label'] == label_id, 'centroid-0'].values,
+        cell_props.loc[cell_props['label'] == label_id, 'centroid-1'].values
+    )).T
+
+    # calculate the total signal intensity within cell
+    cell_counts = EXTRACTION_FUNCTION[extraction](cell_coords, input_images, **kwargs)
+
+    # get morphology metrics
+    current_cell_props = cell_props.loc[cell_props['label'] == label_id, regionprops_names]
+
+    # combine marker counts and morphology metrics together
+    cell_features = np.concatenate((cell_counts, current_cell_props), axis=None)
+
+    # add counts of each marker to appropriate column
+    marker_counts.loc[compartment, cell_id, marker_counts.features[1]:] = cell_features
+
+    # add cell size to first column
+    marker_counts.loc[compartment, cell_id, marker_counts.features[0]] = cell_coords.shape[0]
```

Comment:
I think this is a relic of before we had this nice regionprops set up. Could we include 'area' as a required regionprop? Then we wouldn't need this step

Reply:
`area` is already a required regionprop. If we don't want this `marker_counts.features[0]` assignment, what we would need to do is to remove appending `cell_size` (associated with `ark.settings.PRE_CHANNEL_COL`) as a feature. Is that what we intend?

Reply:
So the current setup always produces a cell table with both area and cell size as separate columns, but we're currently using cell_size as 0th column to determine where the channels start?

Reply:
@ngreenwald I see the problem now. `cell_size` will always be equal to `area`. No need to leave that in then.

Reply:
Yeah, we can change it to label

Reply:
Also I don't get a notification when you edit your comment

Reply:
@ngreenwald gotta keep that in mind for the future.

Reply:
@ngreenwald looking over `transform_expression_matrix` more, it can be a bit of work understanding that `cell_size` and `label` are the 'bread slices' of the channel columns. Two alternatives:

1. Pass in `input_images.channels` as a parameter and use the first and last channel names in this array for indexing.
2. If we can assume that `cell_size` will always the 0th index under the current implementation, then when we remove it, the first channel will become the new 0th index. We then just need the index of `label` to identify the last channel name.

Personally, I like the first one more because it's more flexible to future changes we may make to the cell table structure. However, would definitely like your opinion on this too.

Reply:
You're tlaking about this line? https://github.com/angelolab/ark-analysis/blob/1776cc7647958481c6e0c82c5afeeec473a06608/ark/utils/segmentation_utils.py#L119

Reply:
Yes. So instead, we would do something like:

```
channel_start = np.where(cell_table.features == channels[0])[0][0] + 1
channel_end = np.where(cell_table.features == channels[-1])[0][0] + 1
```

Reply:
I think adding a comment should serve the same purpose of clarification, right? Or is it something about the global variables that makes it particularly confusing?

Reply:
Per today's meeting, we'll open a separate issue/PR for this.